### PR TITLE
Java client bindings

### PR DESCRIPTION
**Pull request by bwalderman**
Adding client language bindings for Java. Sources are found in the java/ directory. The project contains EdgeDriver, EdgeDriverService, and EdgeOptions based on the existing Selenium 3 Chrome* classes.

Selenium 4's Java bindings use separate classes for Edge Legacy and Edge Chromium instead of a single class with a UseChromium switch. Since Selenium 3 already has a driver for Edge Legacy (EdgeHTML), this project simply adds an Edge Chromium driver. The driver that ships with this project is named EdgeDriver for forwards-compatibility with Selenium 4 and because this should hopefully become the default in the future.

TODO: Deploy this to the central Maven repository and/or make available as a downloadable .jar file.

### CODE REVIEW

File: java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+    private String binary;
+    private List<String> args = new ArrayList<>();
+    private List<File> extensionFiles = new ArrayList<>();
+    private List<String> extensions = new ArrayList<>();
+    private Map<String, Object> experimentalOptions = new HashMap<>();
+
+    public EdgeOptions() {
+        setCapability(CapabilityType.BROWSER_NAME, BrowserType.EDGE);
+        setCapability(USE_CHROMIUM, true);
```
Comment (guangyuexu) [inline java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java:93]: Is there an API to set USE_CHROMIUM to false? Do we need a test case for USE_CHROMIUM setup?

File: java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+     * @param acceptInsecureCerts
+     * @return EdgeOptions
+     */
+    public EdgeOptions setAcceptInsecureCerts(boolean acceptInsecureCerts) {
+        setCapability(ACCEPT_INSECURE_CERTS, acceptInsecureCerts);
+        return this;
+    }
+
+    public EdgeOptions setHeadless(boolean headless) {
+        args.remove("--headless");
```
Comment (guangyuexu) [inline java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java:262]: should we remove --disable-gpu when headless is false?

File: java/src/test/java/com/microsoft/edge/seleniumtools/EdgeDriverTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.microsoft.edge.seleniumtools;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class EdgeDriverTest {
```
Comment (guangyuexu) [inline java/src/test/java/com/microsoft/edge/seleniumtools/EdgeDriverTest.java:7]: Can we add some test cases to verify that the right version of edge browser (either legacy edge or chromium edge) is started with the corresponding ms:edgeChromium setup?

File: java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java (hunk 1 above)
Comment (bwalderman) [reply java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java:93]: The Java bindings are Chromium-only unlike the other language bindings. Java users who want to test legacy Edge should continue to use the official Selenium bindings.

This is because of the way the Java EdgeDriver works in Selenium 4 Alpha. They don't have a single class with a useChromium boolean. They have separate EdgeHtmlDriver and EdgeDriver classes. 

So USE_CHROMIUM is always hard-coded to true here.

File: java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java (hunk 2 above)
Comment (bwalderman) [reply java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java:262]: This matches the existing ChromeOptions behavior in Selenium 3. The goal is to provide matching behavior.

I checked if this bug is still present in the latest official Selenium so we can fix it there. It looks like the --disable-gpu option was removed in Selenium 4.0 Alpha 1 so this is no longer a problem: https://github.com/SeleniumHQ/selenium/blob/941dc9c6b2e2aa4f701c1b72be8de03d4b7e996a/py/CHANGES#L87

File: java/src/test/java/com/microsoft/edge/seleniumtools/EdgeDriverTest.java (hunk 3 above)
Comment (bwalderman) [reply java/src/test/java/com/microsoft/edge/seleniumtools/EdgeDriverTest.java:7]: See earlier comment. This EdgeDriver is Chromium-only. I only added minimal tests to verify that Edge Chromium is launched and that ms:edgeOptions is set to true.

**Review by guangyuexu — APPROVED** [review]
